x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2989,7 +3011,16 @@ export class RulesClient {
       });
     }
 
-    await this.taskManager.runSoon(id);
+    try {
+      await this.taskManager.runSoon(attributes.scheduledTaskId ? attributes.scheduledTaskId : id);
```

Comment:
Is there a case where `attributes.scheduledTaskId` is falsy but `id` would actually be valid?  It seems like it would work, but wondering how the rule could get in this state.

Reply:
Hmm...I guess this could happen previously if something occurred when the rule was getting disabled and the rule SO gets updated with a null `scheduledTaskId` but then deleting the associated document ends in an error. If that occurs, the user would be seeing a lot of `Rule ran after disabled` errors.